lib/Remotery.c

```diff
@@ -2263,6 +2293,12 @@ typedef struct ObjectLink_s
     struct ObjectLink_s* volatile next;
 } ObjectLink;
 
+#if defined(RMT_USE_C_ATOMICS)
+    typedef _Atomic(ObjectLink*)    rmtAtomicObjectLinkPtr;
```

Comment:
Nice. I like this. Maybe we can go one better and define a `rmtAtomicPtr(ObjectLink)` macro?